vllm/distributed/kv_transfer/kv_connector/v1/base.py

```diff
@@ -82,12 +83,18 @@ class KVConnectorMetadata(ABC):  # noqa: B024
 
 class KVConnectorBase_V1(ABC):
 
-    def __init__(self, vllm_config: "VllmConfig", role: KVConnectorRole):
+    def __init__(
+        self,
+        vllm_config: "VllmConfig",
+        kv_cache_config: KVCacheConfig,
```

Comment:
These changes to the interface are breaking. We should consider whether we can/should make them in a more backwards-compatible way.

This also turns `KVCacheConfig` which was previously an internal class into part of the public KVConnector interface. We should be sure this is the right thing to do. Is `KVCacheConfig` considered stable (adding fields would be ok I guess).

Also, there's already a similar `register_kv_caches` worker connector method. I feel it may be better to change that method instead to take `KVCacheConfig`. We could introspect the connector impl to check whether it implements this method and the name of the kwarg to determine which to pass.